### ISSUE TITLE
Apply `react-aria` to more components

### ISSUE THREAD

**Issue by ashharrison90** [grafana/grafana#42372]
`react-aria` has some useful hooks we should use in our `@grafana/ui` components (e.g. `useOverlay`/`FocusScope`/`useMenu`, etc.) for consistency. some obvious candidates:

## should be smooth sailing (similar to existing examples) 🥳 
- [x] #42444
- [ ] ~`DateTimePickers` (this is probably getting rewritten, so maybe hold off on this one...)~
- [x] #42571
- [x] #42700

## a little trickier 🤔 
- [x] #42384
- [x] #42383
- [x] #42380

## probably hard 😭 
- [ ] `Menu` (`useMenu`)
- [ ] [Tabs](https://github.com/grafana/grafana/pull/43870) (`useTabList`, `useTab`, `useTabPanel` - see attached draft PR, this would be a breaking change so have decided not to do it for now.
